setup/Setup.CLI/Commands/SetupCommands.cs

```diff
@@ -0,0 +1,77 @@
+using Cocona;
+using Cocona.Application;
+using Terraria.ModLoader.Setup.Core;
+
+namespace Terraria.ModLoader.Setup.CLI.Commands;
+
+public sealed class SetupCommands : BaseCommand
+{
+	private readonly RegenSourceTask.Factory regenSourceTaskFactory;
+	private readonly SetupTask.Factory setupTaskFactory;
+
+	public SetupCommands(
+		RegenSourceTask.Factory regenSourceTaskFactory,
+		SetupTask.Factory setupTaskFactory,
+		ProgramSettings programSettings,
+		TargetsFilesUpdater targetsFilesUpdater,
+		TerrariaExecutableSetter terrariaExecutableSetter)
+		: base(programSettings, targetsFilesUpdater, terrariaExecutableSetter)
+	{
+		this.regenSourceTaskFactory = regenSourceTaskFactory;
+		this.setupTaskFactory = setupTaskFactory;
+	}
+
+	[Command(Description = "Complete environment setup for working on tModLoader source. Equivalent to Decompile + Regen Source")]
+	public async Task Setup(RegenSourceParameters parameters, [FromService] ICoconaAppContextAccessor contextAccessor)
+	{
+		SetupParameters setupParameters = new SetupParameters {
+			NoPrompts = parameters.NoPrompts,
+			PatchMode = parameters.PatchMode,
+		};
+
+		await RunTask(GetSetupOperation(setupParameters), parameters.PlainProgress, parameters.NoPrompts, contextAccessor.Current?.CancellationToken ?? default);
+	}
+
+	[Command(Description = "Setup without any prompts and without parallel decompilation. Intended for e.g. CI pipelines.")]
+	public async Task SetupAuto(SetupParameters parameters, [FromService] ICoconaAppContextAccessor contextAccessor)
+	{
+		if (!string.IsNullOrWhiteSpace(parameters.TerrariaSteamDir)) {
+			ProgramSettings.TerrariaSteamDir = parameters.TerrariaSteamDir;
+		}
+
+		if (!string.IsNullOrWhiteSpace(parameters.TMLDevSteamDir)) {
+			this.ProgramSettings.TMLDevSteamDir = parameters.TMLDevSteamDir;
+			if (!Directory.Exists(parameters.TMLDevSteamDir)) {
+				Directory.CreateDirectory(parameters.TMLDevSteamDir);
+			}
+		}
+
+		ProgramSettings.Save();
```

Comment:
Is it wise to save settings when using `setupAuto`? I know CI pipelines normally clean the entire workspace and do a fresh checkout each time, but for reliable testing I think that `setupAuto` shouldn't read from or write to the settings file at all. It should rely entirely on arguments